chia/wallet/vault/vault_wallet.py

```diff
@@ -0,0 +1,606 @@
+            change_puzzle_hash: bytes32 = next(iter(coins)).puzzle_hash
+            primaries.append(Payment(change_puzzle_hash, uint64(change)))
+
+        spends: List[CoinSpend] = []
+
+        # Check for duplicates
+        all_primaries_list = [(p.puzzle_hash, p.amount) for p in primaries]
+        if len(set(all_primaries_list)) != len(all_primaries_list):
+            raise ValueError("Cannot create two identical coins")
+
+        p2_singleton_puzzle: Program = get_p2_singleton_puzzle(self.vault_info.launcher_coin_id)
+        serialized_puzzle: SerializedProgram = SerializedProgram.from_bytes(bytes(p2_singleton_puzzle))
+
+        for coin in coins:
+            p2_singleton_solution: Program = Program.to([self.vault_info.inner_puzzle_hash, coin.name()])
+            spends.append(
+                CoinSpend(coin, serialized_puzzle, SerializedProgram.from_bytes(bytes(p2_singleton_solution)))
+            )
+
+        return spends
+
+    async def generate_unsigned_vault_spend(
+        self,
+        primaries: List[Payment],
+        p2_spends: List[CoinSpend],
+        memos: Optional[List[bytes]] = None,
+        fee: uint64 = uint64(0),
+        extra_conditions: Tuple[Condition, ...] = tuple(),
+    ) -> Tuple[bytes, Program, Program]:
+        total_amount = (
+            sum(primary.amount for primary in primaries)
+            + fee
+            + sum(c.amount for c in extra_conditions if isinstance(c, CreateCoin))
+        )
+        coins = [spend.coin for spend in p2_spends]
+        spend_value = sum([coin.amount for coin in coins])
+        change = spend_value - total_amount
+        assert change >= 0
+        if change > 0:
+            change_puzzle_hash: bytes32 = get_p2_singleton_puzzle_hash(self.vault_info.launcher_coin_id)
+            primaries.append(Payment(change_puzzle_hash, uint64(change)))
+
+        conditions = [primary.as_condition() for primary in primaries]
+        next_puzzle_hash = await self.get_new_puzzlehash()
+        # TODO: should the vault inner puz create this condition?
+        recreate_vault_condition = CreateCoin(
+            next_puzzle_hash, uint64(self.vault_info.coin.amount), memos=[next_puzzle_hash]
+        ).to_program()
+        conditions.append(recreate_vault_condition)
+        announcements = [CreatePuzzleAnnouncement(spend.coin.name()).to_program() for spend in p2_spends]
+        conditions.extend(announcements)
+
+        delegated_puzzle = puzzle_for_conditions(conditions)
+        delegated_solution = solution_for_conditions(conditions)
+
+        message_to_sign = construct_secp_message(
+            delegated_puzzle.get_tree_hash(),
+            self.vault_info.coin.name(),
+            self.wallet_state_manager.constants.GENESIS_CHALLENGE,
+            self.vault_info.hidden_puzzle_hash,
+        )
+
+        return message_to_sign, delegated_puzzle, delegated_solution
+
+    async def generate_signed_vault_spend(
+        self,
+        signed_message: bytes,
```

Comment:
We don't actually want the paradigm here of some special value passed to some special method.  The paradigm for all wallets now should be the following:
* Call `generate_signed_transaction` (poorly named now it doesn't actually sign anymore) to generate an unsigned transaction for the user
* User calls `gather_signing_info` RPC.  See the following wallet state manager methods to see what you have to implement: https://github.com/Chia-Network/chia-blockchain/blob/603cf479d2b5f14c60bff700694d052c3237142a/chia/wallet/wallet_state_manager.py#L2616-L2650 (it looks like you'll have to extract out the signing target gathering as well)
* User takes signing instructions to signer.  Our wallet is also a signer and in this context the user calls `execute_signing_instructions`.
* User takes signatures and unsigned transaction from previous steps and calls `apply_signatures` RPC
* User takes newly signed transaction and pushes to network